File: src/main/java/myLang/block/statement/IfThenElseStatementContext.java
package myLang.block.statement;

import myLang.ContextHandler;
import myLang.expression.ExpressionContext;
import myLang.response.BooleanResponse;
import myLang.response.MyLangException;
import myLang.response.Response;
import myLangParser.MyLangParser;

public class IfThenElseStatementContext implements ContextHandler<MyLangParser.IfThenElseStatementContext> {
    @Override
    public Response handler(MyLangParser.IfThenElseStatementContext ctx) throws MyLangException {
        Response response = new ExpressionContext().handler(ctx.expression());
        if(!(response instanceof BooleanResponse)){
            throw new MyLangException("Оператор if требует тип boolean");
        }
        if((boolean)response.getResponse()){
            return new StatementContext().handler(ctx.statement(0));
        } else {
            return new StatementContext().handler(ctx.statement(1));
        }
    }
}

File: src/main/kotlin/myLangKotlin/block/statement/IfThenElseStatementContext.kt
package myLangKotlin.block.statement

import myLangKotlin.ContextHandler
import myLangKotlin.expression.ExpressionContext
import myLangKotlin.response.BooleanResponse
import myLangKotlin.response.MyLangException
import myLangKotlin.response.Response
import myLangParser.MyLangParser

class IfThenElseStatementContext : ContextHandler<MyLangParser.IfThenElseStatementContext> {
    @Throws(MyLangException::class)
    override fun handler(ctx: MyLangParser.IfThenElseStatementContext): Response<*> {
        val response = ExpressionContext().handler(ctx.expression()) as? BooleanResponse
                ?: throw MyLangException("Оператор if требует тип boolean")
        return if (response.response as Boolean) {
            StatementContext().handler(ctx.statement(0))
        } else {
            StatementContext().handler(ctx.statement(1))
        }
    }
}
